src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -487,12 +491,7 @@ class BodyScanner extends FilterScanner {
             this.isQuoted = true;
 
             com.sun.tools.javac.util.List<Type> nil = com.sun.tools.javac.util.List.nil();
-            Type quotedOpType = kind == FunctionalExpressionKind.QUOTABLE
-                    ? syms.lambdaOpType
-                    : syms.closureOpType;
-            Type quotedReturnType = new ClassType(null,
-                    com.sun.tools.javac.util.List.of(quotedOpType), syms.quotedType.tsym);
-            MethodType mtype = new MethodType(nil, quotedReturnType, nil, syms.methodClass);
+            MethodType mtype = new MethodType(nil, syms.quotedType, nil, syms.methodClass);
```

Comment:
This code seemed to try to parameterized the `Quoted` type, which is no (longer) a generic type. This was causing a crash in the logic for computing the set of captured variables of a given type (`types::captures`).

This change is what caused the fixes in the two reflect/code tests, as the tests were also expecting a parameterized Quoted type.